src/react-app-env.d.ts

```diff
@@ -1 +1,14 @@
 /// <reference types="react-scripts" />
+
+interface Todos {
+  userId: number | null;
+  id: number;
+  title: string;
+  completed: boolean;
+  user: Users | null;
+};
+
+interface Users {
```

Comment:
ты описываешь один Todo, а не массив
```suggestion
interface Todo {
  userId: number | null;
  id: number;
  title: string;
  completed: boolean;
  user: Users | null;
};

interface User {
```